source/class/qx/tool/cli/Cli.js

```diff
@@ -360,7 +354,7 @@ Versions: @qooxdoo/compiler    v${qx.tool.compiler.Version.VERSION}
             let backup = filepath + ".old";
             await fs.copyFileAsync(filepath, backup);
             if (!this.argv.quiet) {
-              qx.tool.compiler.Console.warn(`*** A backup of ${lockfile} has been saved to ${backup}, in case you need to revert to it. ***`);
+              this.warn(`*** A backup of ${lockfile} has been saved to ${backup}, in case you need to revert to it. ***`);
```

Comment:
This isn't quite right - the point of `this.warn`, `this.debug`, `this.info`, etc is to output messages for developers which can be redirected into different log appenders based on the runtime configuration for logging.  That means that (for example) if you want to debug code inside `qx.tool.cli.commands.Compile` you can increase the logging level for just that class - or you could increase it for `qx.tool.cli.commands.*`.  Or, as you did above on line 134, set the default logging to "debug" and get maximum of output for debugging.

However, these methods are not suitable for interacting with the user, who simply wants to get feedback to better understand what is happening, or have everything quiet unless there is a problem.

As a developer, you might instruct a user to add a logging appender configuration which takes output from `this.info` & `this.warn` etc and direct it to a file.  

When a user asks for output to be "verbose" they are not asking for more debugging info, they're just asking for more feedback.

It is important to be clear about these differences, otherwise code cannot add periodic logging statements with `this.info` and `this.warn` without polluting the console output for the end user - which pushes the developer into using `this.debug` everywhere, which in turn means that the developer's debug logging output is polluted with everything (because `this.debug` is usually reserved for more detailed and code-specific output).

The way to look at it is that console output to the user is "curated" and carefully thought through, but as a developer you want to be focused on how you write code which can be debugged without a debugger (eg because it's running on an user's computer)

This has definitely got muddled over the past few years and you're right that it needs to be cleaned up; there should be a central, easily accessible method to output to the user - eg `qx.tool.compiler.Cli.printVerbose` that only outputs if `--verbose` is set, or maybe `this.printVerbose` for `qx.tool.cli.commands.Command` instances

Each use of `this.warn/debug/info` etc as well as `qx.tool.compiler.Console.warn/info/etc` also needs to be reviewed

Reply:
I fully agree. User feedback has nothing to do with logging. A good indication is if the message in question would make sense to be logged to the development console in the browser, which is clearly not the case here. 

On the contrary, it seems to make sense to have a user interaction API that is agnostic as to the medium by which the message is shown to the user and her feedback is requested - it could be a console or a dialog widget. Maybe we need a new abstract API `qx.io.interaction` with its own  `qx.io.interaction.appender`s? 

But can we do that in a second step? I feel this is outside of the scope of this PR.

Reply:
I think that a new namespace with appenders would be overkill, because there is only likely to be one appender and that's `console.out`.  

The `qx.tool.compiler.Console.warn/info/debug` methods centralise the methods that are called to output to the user, and while they could benefit from detecting whether `--verbose` or `--quiet` are set and suppressing console output automatically, they are adequate.

In our gitter chats, I think I may have guided you down the route of making the changes that you have - apologies if I did, but please can you revert the calls to `qx.tool.compiler.Console.info/warn/etc`

Reply:
I disagree, I think there would be use cases for more than one appender; I also think that a central user-facing message system should be more general than `qx.tool.compiler.Console`. But this is a different question beyond the scope of this PR, and for the purpose of the PR, we should use what we have. I reverted all this.(info|warn|error) and left the debug ones. We will have to look at them carefully whether they are user interaction or log messages later. If this bugs me enough, I'll create an incubator on the basis of which we can continue the discussion.